### DISCUSSION TITLE
Mirth Connect Administrator Launcher - embedded OpenJDK (Azul Zulu Java) updates

### DISCUSSION THREAD

**Discussion by jtgeist-teamhealth** [nextgenhealthcare/connect#6374] [Q&A]
We recently upgraded to the following Mirth software versions

- Mirth Connect 4.5.2.b363
- Mirth Connect Administrator Launcher 1.4.2

Here is our environment system configuration

- Windows Server 2016 (64 bit)
- Eclipse Temurin JDK with Hotspot 17.0.13+ 11 (x64)

Our Tenable scanning software has identified the OpenJDK embedded with the Mirth Connect Administrator Launcher should be updated to the most recent version Azul Zulu: 17.54.21. I had a few questions regarding these findings.

**Is there a way to reconfigure Mirth Connect Administrator Launcher to utilize the system installed Java we have? (Eclipse Temurin JDK)
Or is there a way to update the embedded OpenJDK within the Mirth Connect Administrator Launcher program file location?**

Our security team requires us to update Java on a regular schedule based on Tenable findings and it seems that Mirth Connect Administrator Launcher embeds specific versions which are now out of date for both version 8 and 17.

**Comment by pacmano1** [comment]
Firstly, Tenable would report lots of findings, especially on the server.  But to answer your question on using the system installed one:

You can tell the Mirth Admin Launcher to use the system default as shown in this screenshot.

![image](https://github.com/user-attachments/assets/76310123-bdb6-4f3a-a4b3-49de7c06e5ce)

**Reply by pacmano1** [reply]
I have not tried deleting or updating the JRE installed as part of the mirth admin launcher installer.

**Reply by jtgeist-teamhealth** [reply]
I do see that option on which version of Java to use when launching Mirth Connect from Mirth Connect Administrator Launcher. However, my concern is that Mirth Connect Administrator Launcher installs with a specific version of embedded OpenJDK (Azul Zulu). 

It seems that Mirth Connect Administrator Launcher does not have the option to be installed without the embedded OpenJDK. You are required to select at least version 17 when running the installer for MCAL.

There are multiple plugins from Tenable which have a severity of High based on the version of OpenJDK currently installed by MCAL.
Azul Zulu version installed with MCAL 17.40.19 (released 17-Jan-2023)
Current version for Azul Zulu 17.54.21

I'm trying to determine how I can update Azul Zulu within the embedded version for MCAL or determine if there are options here?

**Reply by pacmano1** [reply]
I just deleted the jre folder and extracted the latest Azul JRE zip to a new jre folder.  It seems to work fine.   You gotta test this stuff yourself.

**Reply by jtgeist-teamhealth** [reply]
Thanks for the help and recommendation! I will attempt this path and confirm if MCAL works as expected on a later version of Azul Zulu JDK.

My concern with this approach is we are now utilizing a "custom" installation for MCAL vs. the ability to utilize the system installed OpenJDK (i.e. C:\Program Files\Eclipse Adoptium). If I was able to use system installed OpenJDK, I wouldn't need to delete and extract the JRE each time into the MCAL folder when MCAL is upgrade/installed, etc.

**Reply by pacmano1** [reply]
If you want Nextgen to update their stuff faster or add a feature, I'd open a ticket on their support portal which is available to paying customers.
